mobile/src/containers/Settings.jsx

```diff
@@ -9,15 +9,16 @@ const mapStateToProps = (state, ownProps) => ({
   version: window.navigator.appInfo ? window.navigator.appInfo.version : 'dev',
   serverUrl: state.mobile.settings.serverUrl,
   backupImages: state.mobile.settings.backupImages,
-  displayUnlinkConfirmation: state.mobile.ui.displayUnlinkConfirmation
+  displayUnlinkConfirmation: state.mobile.ui.displayUnlinkConfirmation,
+  client: state.mobile.settings.client
 })
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
   launchBackup: () => dispatch(mediaBackup()),
   showUnlinkConfirmation: () => dispatch(showUnlinkConfirmation()),
   hideUnlinkConfirmation: () => dispatch(hideUnlinkConfirmation()),
-  unlink: () => {
-    dispatch(unlink())
+  unlink: (client) => {
+    dispatch(unlink(client))
     ownProps.router.replace('/onboarding')
```

Comment:
I don't know if replace is the correct function to use (cf. https://github.com/ReactTraining/react-router/blob/master/docs/API.md#replacepathorloc) or maybe we should replace after clearing the whole navigation history.